src/python/WMCore/Services/RucioConMon/RucioConMon.py

```diff
@@ -0,0 +1,110 @@
+        configDict.setdefault('accept_type', 'application/json')
+        configDict.setdefault('content_type', 'application/json')
+        configDict['logger'] = logger if logger else logging.getLogger()
+        super(RucioConMon, self).__init__(configDict)
+        self['logger'].debug("Initializing RucioConMon with url: %s", self['endpoint'])
+
+    def _getResult(self, uri, callname="", clearCache=True, args=None):
+        """
+        Either fetch data from the cache file or query the data-service
+        :param uri: The endpoint uri
+        :return:    A dictionary
+        """
+
+        # NOTE: Unlike the common case we are not using the callname for building
+        #       apiUrl. we  are using it only for giving proper name of the cache
+        #       file.
+        # NOTE: The 'callname' should not contain '/', otherwise the cache is
+        #       tried to be created with a a non existing subdirectory structure
+
+        cachedApi = "%s.json" % callname
+        # apiUrl = '%s?json&preset=%s' % (uri, callname)
+        apiUrl = uri
+
+        self['logger'].debug('Fetching data from %s, with args %s', apiUrl, args)
+        if args:
+            apiUrl = "%s&%s" % (apiUrl, urlencode(args, doseq=True))
+
+        if clearCache:
+            self.clearCache(cachedApi, args)
+        data = self.refreshCache(cachedApi, apiUrl)
+        results = data.read()
+        data.close()
+
+        results = json.loads(results)
+        return results
+
+    def _getResultZipped(self, uri, callname="", clearCache=True, args=None):
+        """
+        This method is retrieving a zipped file from the uri privided, instead
+        of the normal json
+        :param uri: The endpoint uri
+        :return:    A dictionary
+        """
+
+    def getRSEStats(self):
+        """
+        Gets the latest statistics from the RucioConMon, together with the last
+        update timestamps for all RSEs known to CMS Rucio
+        :param callname:  name of the call
+        :return:          dict of the result
+        """
+        uri = "/WM/stats"
+        rseStats = self._getResult(uri, callname='stats')
+        return rseStats
+
+    def getRSEUnmerged(self, rseName, zipped=False):
+        """
+        Gets the list of all unmerged files in an RSE
+        :param rse: The RSE whose list of unmerged files to be retrieved
+        :return:    A list of unmerged files for the RSE in question
+        """
+        # NOTE: The default API provided by Rucio Consistency Monitor is in a form of a
+        #       zipped file/stream. Currently we are using the newly provided json API
+        #       But in in case we figure out the data is too big we may need to
+        #       implement the method with the zipped API and use disc cache for
+        #       reading/streaming from file. This will prevent any set arithmetic
+        #       in the future.
+        if not zipped:
+            uri = "WM/files?rse=%s&format=json" % rseName
```

Comment:
I would suggest to define the format of the output data to be defined in the `__init__` method, such that it's an object variable. I thing the PhEDEx Service wrapper has something similar to that.

Reply:
I would disagree, though.
This `zipped` interface is a thing that is specific and provided only for this `uri` and nowhere else  in the Rucio Connection Monitor. The reason for that is quite trivial - the files are simply created in .zip format. This is not a real compressed stream. So having it as an object variable would be not only misleading, but also a source of errors.

Reply:
That's a good point! You're right, Todor.